File: Modulo04/Heranca-CSharp/Aluno.cs
using System;

public class Aluno : Pessoa {
    private int ra;

    public Aluno(string nome, int idade, int ra) : base(nome,idade) {
        this.ra = ra;
    }

    public int getRA() {
        return this.ra;
    }
    
    public void setRA(int ra) {
        this.ra = ra;
    }
    
    new public void imprime() {
        base.imprime();
        Console.WriteLine("RA: {0}", this.ra);
    }
}

File: Modulo04/Heranca/Aluno.cpp
#include "Aluno.h"

// Pessoa(nome, idade) <= chamada do construtor da classe base 
Aluno::Aluno(string nome, int idade, int RA) : Pessoa(nome, idade), RA(RA) {
}

int Aluno::getRA() const {
    return RA;
}

void Aluno::setRA(int RA) {
    this->RA = RA;
}

void Aluno::imprime() const {
    Pessoa::imprime();
    cout << "RA: " << RA << endl;
}